### ISSUE TITLE
DSA is broken for some Rust regressions

### ISSUE THREAD

**Issue by shaobo-he** [smackers/smack#503]
Some aliasing pointers are considered non-aliasing by DSA.

**Comment by zvonimir** [comment]
We need a small regression to be able to debug this.

**Comment by keram88** [comment]
This regression demonstrates the problem:
https://github.com/smackers/smack/blob/develop/test/rust/loops/gauss_sum_nondet_fail.rs
It happens here:
`call $p13 := _ZN63_$LT$I$u20$as$u20$core..iter..traits..collect..IntoIterator$GT$9into_iter17hbaa2f2d0c370668fE($i10, $i12);` where `$i12` is loaded from `$M.7`, but in the function body it loads from `$M.12` and `$M.13`.

**Comment by shaobo-he** [comment]
> This regression demonstrates the problem:
> https://github.com/smackers/smack/blob/develop/test/rust/loops/gauss_sum_nondet_fail.rs
> It happens here:
> `call $p13 := _ZN63_$LT$I$u20$as$u20$core..iter..traits..collect..IntoIterator$GT$9into_iter17hbaa2f2d0c370668fE($i10, $i12);` where `$i12` is loaded from `$M.7`, but in the function body it loads from `$M.12` and `$M.13`.

I recall this observation was not accurate.

**Comment by shaobo-he** [comment]
In the regressions, we have the following benchmarks where SMACK reports false alarms,

```
rust/functions/closure.rs
rust/vector/vec1.rs
rust/timeouts/vec_resize.rs
rust/loops/gauss_sum_nondet_fail.rs
```


```

**Comment by shaobo-he** [comment]
This is what I found for `closure.rs`:

in function `_ZN4core3ops8function5impls79_$LT$impl$u20$core..ops..function..FnMut$LT$A$GT$$u20$for$u20$$RF$mut$u20$F$GT$8call_mut17hf0eda75a8d79f977E`, there's a load instruction that uses its argument as the index. However, SMACK assigns a different map for the load instruction and the store instruction in its caller.

**Comment by shaobo-he** [comment]
Fixed via #550
